adapter/utils/src/test/java/org/mobilitydata/gtfsvalidator/timeutils/TimeUtilsImplTest.java

```diff
@@ -240,17 +240,68 @@ void convertHHMMSStoIntegerAndIntegerToHHMMSSShouldBeConsistent() {
     }
 
     @Test
-    void overlappingPeriodsShouldReturnTrue() {
+    void partialOverlapWithOneCommonBoundaryShouldReturnTrue() {
+        //   a |--------|         07:00 am -> 10:00 am
+        //   b     |----|         09:00 am -> 10:00 am
+        // period `b` partially overlaps with period `a`
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(700, 1000,
+                900, 1000));
+
+        // permute periods `a` and `b` in method call
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(900, 1000,
+                700, 1000));
+
+        //   a |--------|         07:00 am -> 10:00 am
+        //   b |----|             07:00 am -> 09:00 am
+        // period `b` partially overlaps with period `a`
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(700, 1000,
+                700, 900));
+
+        // permute periods `a` and `b` in method call
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(700, 900,
+                700, 1000));
+    }
+
+    @Test
+    void partialOverlapPeriodsShouldReturnTrue() {
+        //   a |--------|       07:00 am -> 10:00 am
+        //     b      |----|    08:00 am -> 12:00 am
+        // period `b` partially overlaps with period `a`
         assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(700, 1000,
                 800, 1200));
+
+        // permute periods `a` and `b` in method call
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(800, 1200,
+                700, 1000));
+    }
+
+    @Test
+    void firstPeriodContainedWithTheSecondOneShouldReturnTrueWhenArePeriodsOverlappingIsCalled() {
+        //   a |--------|       07:00 am -> 12:00 am
+        //   b   |----|         08:00 am -> 11:00 am
+        // period `b` is contained within period `a`
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(700, 1200,
+                800, 1100));
     }
 
     @Test
-    void nonOverlappingPeriodsShouldReturnFalse() {
+    void secondPeriodContainedWithinTheFirstOneShouldReturnTrueWhenArePeriodsOverlappingIsCalled() {
+        //   a   |----|         08:00 am -> 11:00 am
+        //   b |--------|       07:00 am -> 12:00 am
+        // period `a` is contained within period `b`
+        assertTrue(TIME_CONVERSION_UTILS.arePeriodsOverlapping(800, 1100,
+                700, 1200));
+    }
+
+    @Test
+    void disjointPeriodsShouldReturnFalse() {
```

Comment:
I think you're still missing the opposite of this test case, where they are disjoint but `b` is before `a`?

Reply:
Oh, nevermind, I see it below. Could you add another comment visual above that test to explain?